pkg/cmd/roachprod/install/expander.go

```diff
@@ -55,26 +64,30 @@ func (e *expander) maybeExpandMap(
 			result = append(result, s)
 		}
 	}
-	return strings.Join(result, " "), true
+	if len(result) != len(nodes) {
+		return "", errors.Errorf("failed to expand nodes %v, given node map %v", nodes, m)
+	}
+	return strings.Join(result, " "), nil
 }
 
-func (e *expander) maybeExpandPgURL(c *SyncedCluster, s string) (string, bool) {
+func (e *expander) maybeExpandPgURL(c *SyncedCluster, s string) (string, bool, error) {
```

Comment:
I'd find a small comment on the retvals helpful. Bool and error is a bit confusing.